soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -3084,6 +3084,98 @@ void Randomizer::CreateCustomMessages() {
+        GIMESSAGE(RG_BOSS_HOUSE_KEY, ITEM_KEY_SMALL,
+            "You found the key to the&%gBoss's House%w!",
+            "Du erhältst einen %rKleinen&Schlüssel%w für das %gBoss-Haus%w!",
+            "Vous obtenez une %rPetite Clé %w&de la %gMaison du Boss%w!"),
+        GIMESSAGE(RG_GRANNYS_POTION_SHOP_KEY, ITEM_KEY_SMALL,
+            "You found the key to&%gGranny's Potion Shop%w!",
+            "Du erhältst einen %rKleinen&Schlüssel%w für den %gTrankladen von Oma%w!",
+            "Vous obtenez une %rPetite Clé %w&du %gMagasin de Potions de Grand-mère%w!"),
+        GIMESSAGE(RG_SKULLTULA_HOUSE_KEY, ITEM_KEY_SMALL,
+            "You found the key to the&%gSkulltula House%w!",
+            "Du erhältst einen %rKleinen&Schlüssel%w für das %gSkulltula-Haus%w!",
+            "Vous obtenez une %rPetite Clé %w&de la %gMaison des Skulltulas%w!"),
+        GIMESSAGE(RG_IMPAS_HOUSE_KEY, ITEM_KEY_SMALL,
+            "You found the key to&%gImpa's House%w!",
+            "Du erhältst einen %rKleinen&Schlüssel%w für das %gHaus von Impa%w!",
+            "Vous obtenez une %rPetite Clé %w&de la %gMaison d'Impa%w!"),
+        GIMESSAGE(RG_WINDMILL_KEY, ITEM_KEY_SMALL,
+            "You found the key to the&%gWindmill%w!",
+            "Du erhältst einen %rKleinen&Schlüssel%w für die %gWindmühle%w!",
+            "Vous obtenez une %rPetite Clé %w&du %gMoulin à Vent%w!"),
+        GIMESSAGE(RG_KAK_SHOOTING_GALLERY_KEY, ITEM_KEY_SMALL,
+            "You found the key to the&%gKakariko Shooting Gallery%w!",
+            "Du erhältst einen %rKleinen&Schlüssel%w für die %gSchießbude von Kakariko%w!",
+            "Vous obtenez une %rPetite Clé %w&du %gStand de Tir de Cocorico%w!"),
+        GIMESSAGE(RG_DAMPES_HUT_KEY, ITEM_KEY_SMALL,
+            "You found the key to&%gDampe's Hut%w!",
+            "Du erhältst einen %rKleinen&Schlüssel%w für %gDampes Hütte%w!",
+            "Vous obtenez une %rPetite Clé %w&du %gChalet de Dampe%w!"),
+        GIMESSAGE(RG_TALONS_HOUSE_KEY, ITEM_KEY_SMALL,
+            "You found the key to&%gTalon's House%w!",
+            "Du erhältst einen %rKleinen&Schlüssel%w für das %gHaus von Talon%w!",
+            "Vous obtenez une %rPetite Clé %w&de la %gMaison de Talon%w!"),
+        GIMESSAGE(RG_STABLES_KEY, ITEM_KEY_SMALL,
+            "You found the key to the&%gStables%w!",
+            "Du erhältst einen %rKleinen&Schlüssel%w für die %gStälle%w!",
+            "Vous obtenez une %rPetite Clé %w&des %gÉcuries%w!"),
+        GIMESSAGE(RG_BACK_TOWER_KEY, ITEM_KEY_SMALL,
+            "You found the key to the&%gBack Tower%w!",
+            "Du erhältst einen %rKleinen&Schlüssel%w für den %gHinteren Turm%w!",
+            "Vous obtenez une %rPetite Clé %w&du %gTour Arrière%w!"),
+        GIMESSAGE(RG_HYLIA_LAB_KEY, ITEM_KEY_SMALL,
+            "You found the key to the&%gHylia Laboratory%w!",
+            "Du erhältst einen %rKleinen&Schlüssel%w für das %gHylia-Labor%w!",
```

Comment:
This one is more or less optional, because `Hylia-Labor` is the correct translation of `Hylia Laboratory`, but neither the German expression nor the English one is a canonical denomination of the location. In English, it is named `Lakeside Laboratory`, in German `Labor am See`, and in French `Laboratoire du Lac`. So my recommendation is to either change all three denominations or change none of them to keep the same expression in all languages.